### DISCUSSION TITLE
Kepler hacks examples

### DISCUSSION THREAD

**Discussion by husky-parul** [sustainable-computing-io/kepler#403] [General]
I would like to propose that we stick to just two environments, OpenShift and Kubernetes when creating manifests for various hacks and poc with Kepler. I understand how tempting it is to try it on different flavors of Kubernetes e.g. `microshift` but having too many manifests will only confuse the users. So the [manifests](https://github.com/sustainable-computing-io/kepler/tree/main/manifests) should have only two varitaions `kubernetes` and `openshift`. That said this is not to discourage community contributions on hacks and poc but we don't want to merge that in kepler instead they can be part of kepler-doc.

Very curious to think what the group think about it.

**Comment by sallyom** [comment]
I have been trying to hack Kepler on MicroShift all week - but what I realize is that the manifests for OpenShift will work just the same on MicroShift - it's the OpenShift manifests we can document officially, agreed. The[ draft PR ](https://github.com/sustainable-computing-io/kepler/pull/376) I have going introduces MicroShift manifests, but I agree they can stay external. Before I move from draft, I'll remove that extra manifests folder and will stick to keeping manifests/openshift up-to-date.

**Reply by rootfs** [reply]
if demo can be recorded, that'll be great

**Comment by marceloamaral** [comment]
As you commented, `microshit` is actually an OpenShift environment. By supporting `OpenShift` and `Kubernetes`, we are already supporting different types of clusters.

**Comment by SamYuan1990** [comment]
hi all,
for any idea, welcome move to [here](https://github.com/sustainable-computing-io/KeplerK8SAction/issues/4) for further discussion.
as we have https://github.com/sustainable-computing-io/KeplerK8SAction I suppose we can reduce the size of hacks folder to avoid duplicate code by sharing or common lib.
